### PR TITLE
MAN-1313: Fix groups not saved when adding spaces

### PR DESCRIPTION
**Pull request by dkinzer**
- [ ] Depends on running `rails data_migration:migrate_space_id` to move `space_id` from `SpaceGroup`  to `Group`.

### CODE REVIEW

**Review by cdoyle-temple — CHANGES_REQUESTED** [review]
@dkinzer Can you retain the order by name in the select list?

File: spec/features/admin/admin_groups_spec.rb [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+
+      visit admin_groups_path + "/#{ group.friendly_id }"
+      expect(page).to have_link(group.space.name)
+
+      visit admin_spaces_path
+      expect(page).to have_link(group.space.name)
+
+      click_link "Destroy"
+      expect(page).not_to have_link(group.space.name)
```
Comment (cdoyle-temple) [inline spec/features/admin/admin_groups_spec.rb:79]: @dkinzer Shouldn't this throw an error that the space is still attached to a model? Or is that something for another ticket?

Comment (dkinzer) [reply spec/features/admin/admin_groups_spec.rb:79]: @cdoyle-temple that's not the current behavior on qa which is why I specifically added the `dependent: :nullify` option to the `has_many groups` directive in the Stage model.  That said if you want it error out we can do that too, but note that currently qa and I assume live simply do not error out when you delete a space that have groups referencing it.

**Comment by dkinzer** [comment]
@cdoyle-temple sorted the the spaces options by name ASC.  Also I added a spec that checks for that so that it's not lost in the future.

**Review by cdoyle-temple — APPROVED** [review]

**Pull request merged**